python/text-to-speech/main.py

```diff
@@ -0,0 +1,293 @@
+"""Synthesize text to speech using Google, Azure and AWS API."""
+# Standard library
+import base64
+import abc
+import json
+
+# Third party
+import requests
+from google.cloud import texttospeech
+import boto3
+
+
+class TextToSpeech():
+    """Base class for Text to Speech."""
+    def __init__(self, req: requests) -> None:
+        """Initialize class method."""
+        self.validate_request(req)
+
+    @abc.abstractmethod
+    def validate_request(self, req: requests) -> None:
+        """Abstract validate request method for providers."""
+
+    @abc.abstractmethod
+    def speech(self, text: str, language: str) -> bytes:
+        """Abstract speech method for providers."""
+
+
+class Google(TextToSpeech):
+    """This class represents the implementation of Google text to speech."""
+    def validate_request(self, req: requests) -> None:
+        """
+        This method validates the request data for Google text to speech.
+
+        Input:
+            req (request): The request provided by the user.
+
+        Raises:
+            ValueError: If any required value is missing or invalid.
+        """
+        if not req.variables.get("API_KEY"):
+            raise ValueError("Missing API_KEY.")
+        if not req.variables.get("PROJECT_ID"):
+            raise ValueError("Missing PROJECT_ID.")
+        self.api_key = req.variables.get("API_KEY")
+        self.project_id = req.variables.get("PROJECT_ID")
+
+    def speech(self, text: str, language: str) -> bytes:
+        """
+        Converts the given text into speech with the Google text to speech API.
```

Comment:
```suggestion
        Convert the given text into speech with the Google text to speech API.
```

Docstrings should be imperative ("Do something ..." or "Return ...") vs descriptive ("Does sometime ..." or "Returns ...").

Same applied to several other docstrings.